src/main/kotlin/org/rust/ide/core/overrideImplement/generateUtil.kt

```diff
@@ -0,0 +1,112 @@
+package org.rust.ide.core.overrideImplement
+
+import com.intellij.codeInsight.generation.ClassMember
+import com.intellij.codeInsight.generation.MemberChooserObject
+import com.intellij.codeInsight.generation.MemberChooserObjectBase
+import com.intellij.ide.util.MemberChooser
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.psi.PsiElement
+import com.intellij.ui.SimpleColoredComponent
+import com.intellij.ui.SimpleTextAttributes
+import io.netty.util.internal.chmv8.ConcurrentHashMapV8
+import org.rust.ide.icons.RsIcons
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsImplItem
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.RsTypeAlias
+import org.rust.lang.core.psi.impl.mixin.default
+import org.rust.lang.core.psi.impl.mixin.isAbstract
+import org.rust.lang.core.psi.util.childOfType
+import org.rust.lang.core.psi.util.trait
+import org.rust.utils.getFormattedParts
+import javax.swing.JTree
+
+sealed class FunctionOrType {
+    class Function(val function: RsFunction) : FunctionOrType()
+    class Type(val type: RsTypeAlias) : FunctionOrType()
+}
+
+private class RsFunctionChooserMember(val base: MemberChooserObjectBase, val member: FunctionOrType) : ClassMember {
+    private val text: String;
+
+    init {
+        when (member) {
+            is FunctionOrType.Function -> {
+                val (before, after) = member.function.getFormattedParts()
+                text = "${member.function.name}$after"
+            }
+            is FunctionOrType.Type -> {
+                text = "${member.type.name}"
+            }
+        }
+    }
+
+    override fun renderTreeNode(component: SimpleColoredComponent?, tree: JTree?) {
+        when (member) {
+            is FunctionOrType.Function -> {
+                component?.append("fn ", SimpleTextAttributes.REGULAR_BOLD_ATTRIBUTES)
+                component?.icon = RsIcons.FUNCTION
+            }
+            is FunctionOrType.Type -> {
+                component?.append("type ", SimpleTextAttributes.REGULAR_BOLD_ATTRIBUTES)
```

Comment:
My arguments are:

- This window shows a part of the project structure tree (ctrl + 7) that relates to a particular trait. So, it needs to follow the same representation rules as the project structure. And neither we nor other plugins I saw show such keywords in the project structure because icons and signatures provide enough information to easily understand what the element is.
- Those keywords are the only bold elements in the tree, so they detract attention a bit from the names of the elements, but are really not important parts of what we want to show here (signatures of items to be implemented). And we also don't bold other keywords in the signatures.
- They introduce indentation for item names, so you need to adjust to it when you're trying to find an element you want to select with eyes. This argument plays together with my next comment that is going to appear shortly :)